src/debuginfo.cpp

```diff
@@ -1139,29 +1156,31 @@ bool jl_dylib_DI_for_fptr(size_t pointer, const llvm::object::ObjectFile **obj,
             debug_link_info info = getDebuglink(*debugobj);
             if (!info.filename.empty()) {
                 size_t sep = fname.rfind('/');
-                ErrorOr<object::OwningBinary<object::ObjectFile>> DebugInfo(std::errc::no_such_file_or_directory);
+#ifdef LLVM39
+                Expected<object::OwningBinary<object::ObjectFile>>
+                    DebugInfo(errorCodeToError(std::make_error_code(std::errc::no_such_file_or_directory)));
+#else
+                ErrorOr<object::OwningBinary<object::ObjectFile>>
+                    DebugInfo(std::errc::no_such_file_or_directory);
+#endif
                 if (fname.substr(sep + 1) != info.filename) {
                     debuginfopath = fname.substr(0, sep + 1);
                     debuginfopath += info.filename;
                     DebugInfo = openDebugInfo(debuginfopath, info);
                 }
-                if (DebugInfo.getError()) {
+                if (!DebugInfo) {
```

Comment:
Should be fixed now. Added a helper function so that there isn't as many `#ifdef`' as there could have been....

The assertion behavior is really confusing. Why does assigning a checked object to another checked object results in an object that has to be checked again?.....................................